.github/workflows/announce_release.yml

```diff
@@ -7,21 +7,47 @@ on:
       - master
     types:
       - completed
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to announce'
+        required: true
 permissions:
+  actions: read
+  contents: read
   issues: write
   pull-requests: write
 
 jobs:
   announce-release-on-released-pull-requests:
     name: Announce release on released pull requests
     runs-on: ubuntu-latest
-    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    if: ${{ github.event.workflow_run.conclusion == 'success' || github.event_name == 'workflow_dispatch' }}
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v6
+      - name: Download release tag artifact
+        if: ${{ github.event_name == 'workflow_run' }}
+        uses: actions/download-artifact@v7
         with:
-          ref: master
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          run-id: ${{ github.event.workflow_run.id }}
+          name: release_tag
+
+      - name: Extract version from artifact
+        if: ${{ github.event_name == 'workflow_run' }}
+        run: |
+          if [ -f "latest.txt" ]; then
```

Comment:
nitpick: I usually prefer fail fast as it reduces amount of code. Do we have such guidelines ?

Reply:
I prefer that too. The earlier iteration had many ifs, but I reworked it then never reduced the logic to fail-fast.